src/MongoDB.Driver.Core/Core/Authentication/ScramShaAuthenticator.cs

```diff
@@ -214,19 +223,34 @@ public ISaslStep Transition(SaslConversation conversation, byte[] bytesReceivedF
                 var nonce = "r=" + r;
                 var clientFinalMessageWithoutProof = channelBinding + "," + nonce;
 
-                var saltedPassword = _hi(
-                    _credential,
-                    Convert.FromBase64String(s),
-                    int.Parse(i));
+                byte[] clientKey;
```

Comment:
Change lines 218-219 above to:

    var salt = Convert.FromBase64String(map['s']);
    var iterations = int.Parse(map['i']);

We're going to need these values in more than one place and might as well Convert/Parse them once.

Reply:
Done